signer/tests/utxo_construction.rs

```diff
@@ -0,0 +1,255 @@
+use signer::utxo::DepositRequest;
+use signer::utxo::SbtcRequests;
+use signer::utxo::SignerBtcState;
+use signer::utxo::SignerUtxo;
+
+mod regtest;
+use regtest::Either;
+use regtest::Recipient;
+
+const SIGNER_ADDRESS_LABEL: Option<&str> = Some("signers-label");
+const DEPOSITS_LABEL: Option<&str> = Some("deposits");
+
+fn make_deposit(
+    depositor: &Recipient,
+    amount: u64,
+    utxo: &ListUnspentResultEntry,
+    signers_public_key: XOnlyPublicKey,
+    faucet_public_key: XOnlyPublicKey,
+) -> (Transaction, DepositRequest) {
+    let fee = regtest::BITCOIN_CORE_FALLBACK_FEE.to_sat();
+    let deposit_script = ScriptBuf::builder()
+        .push_slice([1, 2, 3, 4])
+        .push_opcode(opcodes::all::OP_DROP)
+        .push_opcode(opcodes::all::OP_DUP)
+        .push_opcode(opcodes::all::OP_HASH160)
+        .push_slice(PubkeyHash::hash(&signers_public_key.serialize()))
+        .push_opcode(opcodes::all::OP_EQUALVERIFY)
+        .push_opcode(opcodes::all::OP_CHECKSIG)
+        .into_script();
+
+    let redeem_script = ScriptBuf::new_op_return([0u8, 1, 2, 3]);
+    let ver = LeafVersion::TapScript;
+    let leaf1 = NodeInfo::new_leaf_with_ver(deposit_script.clone(), ver);
+    let leaf2 = NodeInfo::new_leaf_with_ver(redeem_script.clone(), ver);
+
+    let node = NodeInfo::combine(leaf1, leaf2).unwrap();
+    let taproot = TaprootSpendInfo::from_node_info(SECP256K1, faucet_public_key, node);
+    let merkle_root = taproot.merkle_root();
+
+    let deposit_tx = Transaction {
+        version: Version::ONE,
+        lock_time: LockTime::ZERO,
+        input: vec![TxIn {
+            previous_output: OutPoint::new(utxo.txid, utxo.vout),
+            sequence: Sequence::ZERO,
+            script_sig: ScriptBuf::new(),
+            witness: Witness::new(),
+        }],
+        output: vec![
+            TxOut {
+                value: Amount::from_sat(amount),
+                script_pubkey: ScriptBuf::new_p2tr(SECP256K1, faucet_public_key, merkle_root),
+            },
+            TxOut {
+                value: utxo.amount - Amount::from_sat(amount + fee),
+                script_pubkey: depositor.address.script_pubkey(),
+            },
+        ],
+    };
+
+    let (mut tx, signature) = regtest::p2tr_signature(
+        deposit_tx,
+        0,
+        &[utxo.clone()],
+        depositor.keypair,
+        Either::Left(None),
+    );
+    tx.input[0].witness = Witness::p2tr_key_spend(&signature);
+
+    let req = DepositRequest {
+        outpoint: OutPoint::new(tx.compute_txid(), 0),
+        max_fee: fee,
+        signer_bitmap: Vec::new(),
+        amount: 25_000_000,
+        deposit_script: deposit_script.clone(),
+        redeem_script: redeem_script.clone(),
+        taproot_public_key: faucet_public_key,
+        signers_public_key,
+    };
+    (tx, req)
+}
+
+/// This test just checks that many of the methods on the Recipient struct
+/// work as advertised.
+#[test]
+#[cfg_attr(not(feature = "integration-tests"), ignore)]
```

Comment:
Nit: Instead of individually feature flagging the tests, we could have a single cfg attribute for the whole module, saving us some bytes in the code and some compile time when building and running non-integration tests.

Reply:
This turns out to be a little more difficult to do than I would like. You have to use `cfg` on the module, but then rust-analyzer doesn't analyze the module by default since it doesn't check for all features by default. You can change it so that rust-analyzer picks up any or all features, but then I'd like to get more buy-in from the team for such a change. For now, keeping the `cfg_attr` on each test is the simplest solution so I'd like to keep it like this until we are all on board for such a change.

I did change things so that they are now under an `integration` folder. If all tests are in that folder, then this will limit the number of test binaries that cargo builds during `cargo test` to one.

Reply:
Right, that makes sense. Thank you for trying it out at least. I'm happy to leave this as-is :)